app/controllers/api/v1/transactions_controller.rb

```diff
@@ -52,7 +52,7 @@ def transaction_params
           debtor: t[:debtor] ? User.find_by(name: t[:debtor]) : User.zeus,
           creditor: t[:creditor] ? User.find_by(name: t[:creditor]) : User.zeus,
           issuer: current_user_or_client,
-          amount: ((BigDecimal(t[:euros] || 0, 2) * 100) + t[:cents].to_i).to_i,
+          amount: ((BigDecimal(t[:euros]&.to_s || "0") * 100) + t[:cents].to_i).to_i,
```

Comment:
In the case that `t[:euros]` is a float I think it would be better to format it using `sprintf("%.2f", t[:euros])` to avoid any weird floating point errors. E.g.:
```
irb(main):009> (BigDecimal((0.3 - 0.2).to_s) * 100).to_i
=> 9
```
vs.
```
irb(main):010> (BigDecimal(sprintf("%.2f", (0.3 - 0.2))) * 100).to_i
=> 10
```